### ISSUE TITLE
Render Transform Override appears to mess with motion vectors / motion blur in cameras

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#1654]
### Describe the bug?

When looking at the default head and hands avatar through a camera with motion blur enabled, the head looks very fuzzy.

### To Reproduce

Spawn the default streaming camera and equip the default avatar. Notice how the head appears fuzzy in the camera render (but not on photos).

### Expected behavior

The head not to appear fuzzy in cameras.

### Screenshots

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/6e5d6be6-84be-4aa8-b89e-192fac7fd446)

### Resonite Version Number

Beta 2024.4.3.1170

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Valve Index

### Log Files

N/A

### Additional Context

I noticed this in two YouTube recordings of new users now. I also have the same issue with the hair on my avatar which I hide with RTO. A temporary workaround for the content team could be disabling the RTO and instead adjusting the near clip distance, at least until the underlying issue is resolved. (RenderMaterialOverride won't work for the eyes, as they are canvases)

https://www.youtube.com/watch?v=rjYO6QAEtVo
https://www.youtube.com/watch?v=T_l0WXQ2hkw

**Comment by Dusty-Sprinkles** [comment]
If this counts as a bug would the other RTO quirks (not being able to see items parented to an overwritten slot in a mirror unless its at pretty much 1 scale for one) also would they fit under this issue or need its own?

**Comment by shininghero** [comment]
Oh, this has been around for a while.  I assumed it was a quick bandaid to mitigate crashing while they gear up for a proper fix.
...I think?  Could be wrong about that.

I do know for certain that the main RTO crash ticket (#269) is blocked, pending a potential Unity minor version upgrade (#585).  But that has its own rendering issue holding it up.

**Comment by Electronus** [comment]
I noticed this happens when you don't fill in the skinned mesh renderer in the component. Filling that out fixed this issue as far as I can tell.

**Comment by Frooxius** [comment]
This is not really a content issue, it's a problem with RenderTransformOverride, which technically moves the object and generates motion vectors - this is not an intended behavior.

The content team could remove the component, but that's not really fixing the bug, it's just a workaround.

**Comment by Frooxius** [comment]
@shiftyscales Yes. You only need to assign renderer if it's a SkinnedMeshRenderer, because it needs to specifically flag those to recompute the skinning on the override.

With static MeshRenderers there's no recompute needed - they can just be naively moved, so they don't need to be references.

**Comment by AegisTheWolf** [comment]
The confusion at first was why it isnt on the one in the essentials but found the component in the spawn objects, now it is removed on the one in spawn objects once its uploaded, this issue should be fixed soon

**Comment by JackTheFoxOtter** [comment]
@AegisTheWolf as of today (Beta 2024.4.26.10), the default avatar from the spawn objects still has the blur issue in cameras. The one in the Resonite Essentials folder is fixed.

**Comment by sveken** [comment]
I think due to the new update adding rto to everyone. this now happens all the time

**Comment by ko-tengu** [comment]
Couldn't this be at least mitigated by just disabling motion blur in the camera preview?

I was thinking about it again because the camera I normally use does this (orange's depthcam) and was distracted by a default camera someone spawned in a session.

**Comment by rabbuttz** [comment]
Many of my friends are also experiencing this issue. When streamers and other content creators came to Resonite to broadcast, I helped support them, but this problem occurred. If there is a solution or even a temporary workaround, it would be very helpful.

**Comment by Frooxius** [comment]
Since the renderer is now open source, if anyone would like to poke around this and figure out a good solution (e.g. suppressing some transform updates from motion vectors), feel free!

https://github.com/Yellow-Dog-Man/Renderite.Unity.Renderer
